### PR TITLE
Add docstrings for functions

### PR DESCRIPTION
**Pull request by jkbest2**
Let me know if I missed anything. I tried to follow [BlueStyle](https://github.com/invenia/BlueStyle#documentation). I can try to write docstrings for the rest of the functions next time I have a chance.

### CODE REVIEW

File: src/spde_precision_matrices.jl [hunk 1]
```diff
@@ -1,4 +1,16 @@
+"""
+    component_matrices(mesh::TriMesh, κ::real) -> (C::Matrix, C̃::Matrix, G::Matrix)
 
+From a triangulated mesh of a 2-dimensional domain, generate the finite element
+method matrices C, C̃, and G as described in Lindgren et al. 2011.
+
+# Arguments
+- `mesh::TriMesh`: a mesh from TriangleMesh.jl
+- `κ::Real`: Matérn smoothness parameter
```
Comment (ElOceanografo) [inline src/spde_precision_matrices.jl]: This should be "Matérn correlation decay rate parameter"

**Review by ElOceanografo — CHANGES_REQUESTED** [review]
Looks good, just one comment.  Thanks for writing these!

File: src/spde_precision_matrices.jl (hunk 1 above)
Comment (jkbest2) [reply src/spde_precision_matrices.jl]: Fixed below

**Comment by jkbest2** [comment]
Added docstrings for the mesh utility functions and changed the description of κ as requested. I think I got the mesh function descriptions correct, but let me know if there are any more changes.

File: src/mesh_utils.jl [hunk 2]
```diff
@@ -13,7 +13,20 @@
 # end
 #
 
+"""
+    make_border_points(points, n, expansion, k = size(points, 2)) -> Matrix
 
+Determine points along the edge of the domain.
```
Comment (ElOceanografo) [inline src/mesh_utils.jl]:
```suggestion
Create a set of border points around the hull of a domain.
```

File: src/mesh_utils.jl [hunk 3]
```diff
@@ -13,7 +13,20 @@
...
+
+# Arguments
+- `points`: points to form a hull around
```
Comment (ElOceanografo) [inline src/mesh_utils.jl]:
```suggestion
- `points`: points to form a hull around, as a matrix with a point in each column
```

File: src/mesh_utils.jl [hunk 4]
```diff
@@ -13,7 +13,20 @@
...
+- `n`: number of edge points
+- `expansion`: factor to extend the hull beyond the provided points
+- `k`: number of neighbors to use when determining hull; larger is smoother
```
Comment (ElOceanografo) [inline src/mesh_utils.jl]:
```suggestion
- `k`: number of neighbors to use when determining hull. Default is all points, giving a convex hull; can be decreased to make the hull concave.
```

File: src/mesh_utils.jl [hunk 5]
```diff
@@ -38,11 +51,34 @@ function make_border_points(points, n, expansion, k=size(points, 2))
...
+"""
 function inverse_distance_weights(dists::AbstractVector)
     w = [d > 0 ? 1/d : 1/(d+eps()) for d in dists]
     return w / sum(w)
 end
 
+"""
+    observation_matrix(mesh::TriMesh, points::AbstractMatrix) -> SparseMatricCSC
+
+Calculate the weights to interpolate from nodes to observation locations.
```
Comment (ElOceanografo) [inline src/mesh_utils.jl]:
```suggestion
Create matrix of weights to linearly interpolate from nodes to observation locations.
```

**Review by ElOceanografo — CHANGES_REQUESTED** [review]
Looking good, just some minor wording edits

**Comment by jkbest2** [comment]
Thanks for the edits; I applied them. This should be ok to merge now.

**Review by ElOceanografo — APPROVED** [review]

**Comment by ElOceanografo** [comment]
Looks good, thanks!

**Pull request merged**